core/bridges/orm.go

```diff
@@ -198,7 +191,36 @@ func (o *orm) UpsertBridgeResponse(ctx context.Context, dotId string, specId int
 				DO UPDATE SET value = $3, finished_at = $4;`
 
 	_, err := o.ds.ExecContext(ctx, sql, dotId, specId, response, time.Now())
-	return pkgerrors.Wrap(err, "failed to upsert bridge response")
+
+	return err
+}
+
+func (o *orm) BulkUpsertBridgeResponse(ctx context.Context, responses []BridgeResponse) error {
+	argCnt := 4
+	valueStrings := make([]string, len(responses))
+	valueArgs := make([]interface{}, argCnt*len(responses))
+
+	for idx, resp := range responses {
+		valueStrings[idx] = "(?, ?, ?, ?)"
+
+		pointIdx := argCnt * idx
+
+		valueArgs[pointIdx] = resp.DotId
+		valueArgs[pointIdx+1] = resp.SpecId
+		valueArgs[pointIdx+2] = resp.Response
+		valueArgs[pointIdx+3] = resp.FinishedAt
```

Comment:
Isn't there a way to pass a slice of structs to insert? Like with named arguments?

Reply:
I found one with `sqlx` package. That's cleaner.